components/api-resource-mgt/org.wso2.carbon.identity.api.resource.collection.mgt/src/main/java/org/wso2/carbon/identity/api/resource/collection/mgt/util/FilterUtil.java

```diff
@@ -0,0 +1,200 @@
+            }
+            setExpressionNodeList(node.getLeftNode(), expression);
+            setExpressionNodeList(node.getRightNode(), expression);
+        }
+    }
+
+    /**
+     * Filter API resource collections.
+     *
+     * @param apiResourceCollectionsMap API resource collection map.
+     * @param filter                    Filter string.
+     * @return Map of API resource collection.
+     * @throws APIResourceCollectionMgtClientException Error when validate filters.
+     */
+    public static Map<String, APIResourceCollection> filterAPIResourceCollections(
+            Map<String, APIResourceCollection> apiResourceCollectionsMap, String filter)
+            throws APIResourceCollectionMgtClientException, APIResourceCollectionMgtServerException {
+
+        try {
+            List<ExpressionNode> expressionNodes = getExpressionNodes(filter);
+            return apiResourceCollectionsMap.entrySet().stream()
+                    .filter(entry -> {
+                        try {
+                            return matchesFilter(entry.getValue(), expressionNodes);
+                        } catch (APIResourceCollectionMgtClientException e) {
+                            throw new RuntimeException(e);
+                        }
+                    })
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        } catch (RuntimeException e) {
+            if (e.getCause() instanceof APIResourceCollectionMgtClientException) {
+                throw (APIResourceCollectionMgtClientException) e.getCause();
+            } else {
+                throw handleServerException(
+                        APIResourceCollectionManagementConstants.ErrorMessages
+                                .ERROR_CODE_WHILE_FILTERING_API_RESOURCE_COLLECTIONS, e);
+            }
+        }
+    }
+
+    /**
+     * Match the filter with the API resource collection.
+     *
+     * @param apiResourceCollection API resource collection basic info object.
+     * @param expressionNodes       List of expression nodes.
+     * @return True if the filter matches with the API resource collection object.
+     * @throws APIResourceCollectionMgtClientException Error when validate filters.
+     */
+    private static boolean matchesFilter(APIResourceCollection apiResourceCollection,
+                                         List<ExpressionNode> expressionNodes)
+            throws APIResourceCollectionMgtClientException {
+
+        for (ExpressionNode node : expressionNodes) {
+            String attribute = node.getAttributeValue();
+            String value = node.getValue();
+            String operation = node.getOperation();
+            String apiResourceCollectionAttributeValue = getAttributeValue(apiResourceCollection, attribute);
+            if (apiResourceCollectionAttributeValue == null) {
+                return false;
+            }
+            switch (operation) {
```

Comment:
Instead of having multiple if conditions, shall we refactor the code as provided below.

```
boolean matches;
switch (operation) {
    case APIResourceCollectionManagementConstants.EQ:
        matches = apiResourceCollectionAttributeValue.equals(value);
        break;
    case APIResourceCollectionManagementConstants.CO:
        matches = apiResourceCollectionAttributeValue.contains(value);
        break;
    case APIResourceCollectionManagementConstants.SW:
        matches = apiResourceCollectionAttributeValue.startsWith(value);
        break;
    case APIResourceCollectionManagementConstants.EW:
        matches = apiResourceCollectionAttributeValue.endsWith(value);
        break;
    default:
        throw handleClientException(
                APIResourceCollectionManagementConstants.ErrorMessages.ERROR_CODE_INVALID_FILTER_OPERATION);
}

if (!matches) {
    return false;
}
```

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/5807.